### PR TITLE
Add auth with  google API and passport

### PR DESCRIPTION
**Pull request by andrewzb**
Add bd associations

### CODE REVIEW

File: config/postgresConnection.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "development": {
-    "username": "username",
-    "password": "password",
-    "database": "database",
+    "username": "postgres",
+    "password": "1111",
+    "database": "passport",
```
Comment (Dmitriyjd) [inline config/postgresConnection.json]: Do not change this file because this is not secure. This file should contain only dummy values

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "3.0.0",
   "license": "MIT",
   "scripts": {
-    "start": "cross-env PORT=4000 razzle start",
+    "start": "cross-env razzle start",
```
Comment (Dmitriyjd) [inline package.json]: We need to support any port to prevent possible port conflicts.

File: src/common/App.js [hunk 3]
```diff
@@ -2,14 +2,14 @@ import React from 'react';
 import { Route, Switch } from 'react-router-dom';
 import Home from 'common/pages/home/containers';
 import Edit from 'common/pages/edit/containers';
-
+import Start from './pages/start';
```
Comment (Dmitriyjd) [inline src/common/App.js:6]: Imports from external libraries should be before your custom code.
Correct order: 
import { Route, Switch } from 'react-router-dom';
import 'antd/dist/antd.css';
import Start from './pages/start';
import Home from 'common/pages/home/containers';
import Edit from 'common/pages/edit/containers';

File: src/common/pages/start/component.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  StyledLoginLink,
+  StyledLogoutLink,
+} from './styles';
+
+const Start = () => (
+  <StyledMainWrapper>
+    <StyledButtonsWrapper>
+      <StyledLoginLink href="auth/google">
+        <FontAwesomeIcon icon={faGooglePlus} />
+        Authorize with Gooogle +
```
Comment (Dmitriyjd) [inline src/common/pages/start/component.jsx]: Wrap this text into `<span/>` or `<div/>`

File: src/common/pages/start/styles.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import styled from 'styled-components';
+
+export const StyledMainWrapper = styled.div`
+  position: relative;
+  height: 100vh;
```
Comment (Dmitriyjd) [inline src/common/pages/start/styles.js:18]: Try to prevent using `vh`. Try to use `%` instead.

File: src/common/pages/start/styles.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import styled from 'styled-components';
+
+export const StyledMainWrapper = styled.div`
+  position: relative;
```
Comment (Dmitriyjd) [inline src/common/pages/start/styles.js]: Don't use `position: relative`. Try to use flexbox or css grid instead.

File: src/common/pages/start/styles.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    #207cca 32%,
+    #207cca 32%,
+    #2989d8 50%,
+    #1e5799 97%,
+    #7db9e8 100%
+  );
+`;
+
+export const StyledButtonsWrapper = styled.div`
+  position: absolute;
```
Comment (Dmitriyjd) [inline src/common/pages/start/styles.js]: Don't use `position: absolute`. Try to use flexbox or css grid instead.

File: src/common/pages/start/styles.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  padding: 8px 20px;
+  text-decoration: none;
+  font-weight: bold;
+  border-radius: 5px;
+  cursor: pointer;
+  font-size: 1.5rem;
+`;
+
+export const StyledLoginLink = styled(StyledLink)`
+  background-color: #00d4ff;
```
Comment (Dmitriyjd) [inline src/common/pages/start/styles.js]: Try to move colors into constants.

File: src/server/config/database-config.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import Sequelize from 'sequelize';
+
+export default new Sequelize('passport', 'postgres', '1111', {
+  host: 'localhost',
+  dialect: 'postgres',
+});
```
Comment (Dmitriyjd) [inline src/server/config/database-config.js]: 1. This is not a config. This file creates connection to database. Move it into utils folder and rename it into `CreateDBConnection.js`.
2. Use CamelCase instead kebab-case.
3. Use Config from `config/postgresConnection.json`, because we need single source of trust for database config.

File: src/server/config/keys.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+export default {
+  google: {
+    clientID:
+      // eslint-disable-next-line max-len
+      '294540986299-on4oith7fml2vhsumdesfr9npjr8r2um.apps.googleusercontent.com',
+    clientSecret: 'Iasf-PGRAwt6MhDZpuzI2Ems',
```
Comment (Dmitriyjd) [inline src/server/config/keys.js]: There are secure keys. We don't need to push it into public repository. Pls create dummy keys insteal and share those values using private conference.

File: src/server/config/passport-setup.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
+import passport from 'passport';
+import GoogleStrategy from 'passport-google-oauth20';
+import keys from './keys';
+import DB from '../models/index';
```
Comment (Dmitriyjd) [inline src/server/config/passport-setup.js]: You can use `import DB from '../models';` instead

File: src/server/config/passport-setup.js [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+const { User } = DB;
+
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((id, done) => {
+  User.findByPk(id).then(user => {
+    done(null, user);
+  });
```
Comment (Dmitriyjd) [inline src/server/config/passport-setup.js]: Move this statement to User's model functions, like `findUserById()`.

File: src/server/config/passport-setup.js [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+});
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: keys.google.clientID,
+      clientSecret: keys.google.clientSecret,
+      callbackURL: '/auth/google/redirect',
```
Comment (Dmitriyjd) [inline src/server/config/passport-setup.js]: Move this string to config file. Put it near Google keys.

File: src/server/config/passport-setup.js [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+          return User.create({
+            googleId: profile.id,
+            email: profile.emails[0].value,
+            name: profile.displayName,
+            img: profile.photos[0].value,
+          });
+        })
+        .then(user => {
+          done(null, user);
+        });
```
Comment (Dmitriyjd) [inline src/server/config/passport-setup.js]: Move this statements to User's model functions, like `findUserByGoogleId()` and `createUser()`.

File: src/server/config/passport-setup.js [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+    },
+  ),
+);
```
Comment (Dmitriyjd) [inline src/server/config/passport-setup.js]: 1. This is not a config.
2. Use CamelCase instead kebab-case.

File: src/server/index.js [hunk 16]
```diff
@@ -1,18 +1,42 @@
...
+
+import './config/passport-setup';
 
 const server = express();
+server.use(bodyParser.urlencoded({ extended: true }));
+server.use(bodyParser.json());
+
+server.use(
+  cookieSession({
+    maxAge: 24 * 60 * 60 * 1000,
```
Comment (Dmitriyjd) [inline src/server/index.js]: Move this statement into constants file.

File: src/server/index.js [hunk 17]
```diff
@@ -1,18 +1,42 @@
...
+    keys: [keys.session.cookieKey],
+  }),
+);
+
+server.use(passport.initialize());
+server.use(passport.session());
+server.use('/auth', userOauthRouth);
```
Comment (Dmitriyjd) [inline src/server/index.js:23]: Include REST API controllers for users and passwords

File: src/server/models/user.js [hunk 18]
```diff
@@ -1,51 +1,43 @@
...
-// because this file is imported by sequelize-cli(without babel)
-const Sequelize = require('sequelize');
-const postgresConnection = require('../../../config/postgresConnection');
+import Sequelize from 'sequelize';
+import db from '../config/database-config';
 
-const { Model } = Sequelize;
+const defaultUserImg =
+  // eslint-disable-next-line max-len
+  'https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwjp7uSztJ3jAhWBjKQKHSZwAqkQjRx6BAgBEAQ&url=https%3A%2F%2Fdeathbattlefanon.fandom.com%2Fwiki%2FChin-Chin&psig=AOvVaw3O0n5ETwjexM5gl5avRJsb&ust=1562403407133663';
```
Comment (Dmitriyjd) [inline src/server/models/user.js]: Move this image into assets.

File: src/server/models/user.js [hunk 19]
```diff
@@ -1,51 +1,43 @@
...
+      allowNull: false,
+      defaultValue: defaultUserImg,
+    },
   },
-  updatedAt: {
-    allowNull: false,
-    type: Sequelize.DATE,
+  {
+    // eslint-disable-next-line max-len
+    timestamps: false, // Колонки createdAt и updatedAt будут созданы автоматически
```
Comment (Dmitriyjd) [inline src/server/models/user.js]: 1. We need to have to createdAt and updatedAt to have a consistancy between model and existing database.
2. Avoid russian letters, because we need to keep our code extendable.

File: src/server/routes/userOauth.js [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+  res.redirect('/');
+});
+
+// auth with google+
+router.get(
+  '/google',
+  passport.authenticate('google', {
+    scope: [
+      'https://www.googleapis.com/auth/userinfo.profile',
+      'https://www.googleapis.com/auth/userinfo.email',
```
Comment (Dmitriyjd) [inline src/server/routes/userOauth.js]: Move this statement into constants.

File: src/server/routes/userOauth.js [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+
+// callback route for google to redirect to
+// hand control to passport to use code to grab profile info
+router.get('/google/redirect', passport.authenticate('google'), (req, res) => {
+  const { user } = req;
+
+  let msg = 'your are not autinficated';
+  if (user) {
+    // eslint-disable-next-line max-len
+    msg = `= взято из req.user после отработки кукей == username - ${user.name}||img-- ${user.img} || email- ${user.email}||googleId - ${user.googleId}`;
```
Comment (Dmitriyjd) [inline src/server/routes/userOauth.js:32]: How do you feel, do we need to refactor it?

File: src/server/models/password.js [hunk 22]
```diff
@@ -1,91 +1,32 @@
...
-
-module.exports = {
-  passwordSchema,
-  Password,
-  createPassword,
-  getPasswordById,
-  getPasswordsByUserId,
-  deletePasswordById,
-  updatePasswordById,
-};
```
Comment (Dmitriyjd) [inline src/server/models/password.js]: Why did you remove REST API logic?

File: src/server/models/index.js [hunk 23]
```diff
@@ -0,0 +1,19 @@
...
+Password.belongsTo(User, { foreignKey: 'UserId' });
+DB.User = User;
+DB.Password = Password;
+
+// eslint-disable-next-line max-len
+//  if you need to drop tours tables and reset then run this code ones(coment in => run => coment out)
+// sequelize
+// .sync({force:true})
+// .then(s => console.log('Синхронизировано'))
+// .catch((err) => console.log(err));
```
Comment (Dmitriyjd) [inline src/server/models/index.js:18]: Move it into separate file and add possibility via npm script.

File: src/server/migrations/create-user.js [hunk 24]
```diff
@@ -1,6 +1,6 @@
-const { userSchema } = require('../models/user');
+const { user } = require('../models/user');
 
 module.exports = {
-  up: queryInterface => queryInterface.createTable('Users', userSchema),
-  down: queryInterface => queryInterface.dropTable('Users'),
+  up: queryInterface => queryInterface.createTable('User', user),
+  down: queryInterface => queryInterface.dropTable('User'),
 };
```
Comment (Dmitriyjd) [inline src/server/migrations/create-user.js]: Use CamelCase for file names instead of kebab-case

**Review by Dmitriyjd — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (andrewzb) [reply package.json]: in cross origin resource sharing port:3000 & port:4000 treated as difrent origins

**Comment by dzianisbohush** [comment]
this changes in another PR
